app/src/pages/playground/playgroundUtils.ts

```diff
@@ -948,44 +950,31 @@ export const getToolName = (tool: Tool): string | null => {
  * Creates a tool definition for the given provider
  * @param provider the provider to create the tool for
  * @param toolNumber the tool number to create - used for naming the tool
- * returns a tool definition for the given provider
+ * @param type the type of the tool
+ * @param definition the definition of the tool. In OpenAI format, will be converted to the appropriate format for the provider.
+ * @returns a tool definition for the given provider
  */
 export const createToolForProvider = ({
   provider,
   toolNumber,
+  type = "json",
+  definition,
 }: {
   provider: ModelProvider;
   toolNumber: number;
+  type?: PhoenixToolTypeType;
+  definition?: OpenAIToolDefinition;
 }): Tool => {
-  switch (provider) {
-    case "OPENAI":
-    case "DEEPSEEK":
-    case "XAI":
-    case "OLLAMA":
-    case "AZURE_OPENAI":
-      return {
-        id: generateToolId(),
-        definition: createOpenAIToolDefinition(toolNumber),
-      };
-    case "ANTHROPIC":
-      return {
-        id: generateToolId(),
-        definition: createAnthropicToolDefinition(toolNumber),
-      };
-    case "AWS":
-      return {
-        id: generateToolId(),
-        definition: createAwsToolDefinition(toolNumber),
-      };
-    // TODO(apowell): #5348 Add Google tool definition
-    case "GOOGLE":
-      return {
-        id: generateToolId(),
-        definition: createOpenAIToolDefinition(toolNumber),
-      };
-    default:
-      assertUnreachable(provider);
-  }
+  const defaultDefinition = fromOpenAIToolDefinition({
+    toolDefinition: definition ?? createOpenAIToolDefinition(toolNumber),
+    targetProvider: provider,
```

Comment:
I'm not sure I follow how this is backwards compatible given the switch statement in the diff